emeraldwalk/vscode-runonsave
# Add notebook document support

## Description of Changes
This change adds support for notebooks, such as those used by Jupyter Notebooks (`NotebookDocument`), by utilizing `onDidSaveNotebookDocument`  as `onDidSaveTextDocument` is not triggered when a notebook is saved.

References to `document.fileName` have been replaced with `document.uri.fsPath` using a Union type to support both types, as `NotebookDocument `does not have the equivalent `fileName` attribute.

This PR also includes a small update to `launch.json` to disable extensions during development testing.

Closes #57

## Testing
This extension does not currently have a test suite, but the functionality was tested manually for both text documents and notebook documents.

Testing Steps:
- Create a notebook file and text file.
- Use an application configuration such as:
    ```jsonc
    {
		"emeraldwalk.runonsave": {
			"commands": [
				{
					// Run whenever any file is saved
					"match": ".*",
					"cmd": "echo '${fileBasename}' saved."
				}
			]
		}
    }
    ```
- Save with both types and ensure correct output

## Checklist
- [✔️] Did you update any relevant docs / samples?
- [✔️] Did you include details on how to test your changes?
